shim/stub_test.go

```diff
@@ -235,6 +236,19 @@ func TestChaincodeStubGetTxTimestamp(t *testing.T) {
 	}
 }
 
+func TestGetMSPID(t *testing.T) {
+	_, err := GetMSPID()
+	assert.EqualError(t, err, "'CORE_PEER_LOCALMSPID' is not set")
+
+	os.Setenv("CORE_PEER_LOCALMSPID", "mspid")
+
+	mspid, err := GetMSPID()
+	assert.NoError(t, err)
+	assert.Equal(t, "mspid", mspid)
+
+	os.Unsetenv("CORE_PEER_LOCALMSPID")
```

Comment:
Sort of a nit: If you're going to change the environment, ideally you restore it at test exit. This code is blindly setting and unsetting without checking for an existing value.

This only becomes a problem in, pretty much, every single test scenario that involves viper...

Reply:
Stepping back, you are failing the test if the variable is set... So you're not changing the environment, you're just depending on that var never being set when running. Smaller scope.

Reply:
Makes sense, will consider for future reference